lib/apiml.js

```diff
@@ -280,7 +281,12 @@ ApimlConnector.prototype = {
   },
 
   getServiceUrls() {
-    return this.discoveryUrls.map(url => url + (url.endsWith('/') ? '' : '/') + 'apps');
+    let urls = this.discoveryUrls.map(url => url + (url.endsWith('/') ? '' : '/') + 'apps');
+    if (this.isClientAttls) {
+      return urls.map(url => url.replaceAll('https', 'http'));
+    } else {
+      return urls;
+    }
   },
 
   getRequestOptionsArray(method, path) {
```

Comment:
Hi Sean, great to see that, yes i've come up to the same results, that app server is still trying to use https to contact Eureka in at-tls, was hoping to address that with different pagent rules, wouldn't be correct at-tls i just wanted to find a workaround. 
Anyway, i'll be able to test this PR only after holidays, but there a few things that are pop up in my head from previous tests:
The function below (getRequestOptionsArray) uses hardcoded 8 characters to slice for "https://" and so it will cut one char from host name in case of "http://" 
Also a minor schema thing, we allow using custom URL for discoveryUrls in zowe.yaml
`app-server.node.mediationLayer.server.discoveryUrls`
However schema will expect https there and will break if we use http url there.

Reply:
discoveryUrls comes from the discovery env var unless from this override of `app-server.node.mediationLayer.server.discoveryUrls`
in that env var, it's always "https://"
`getServiceUrls()` is now returning URLs that could be https or http.
But `getRequestOptionsArray()` isnt transforming the input, it's just extracting host & port. So I think it's okay that it's hardcoded to `https://` still.
I also think `app-server.node.mediationLayer.server.discoveryUrls` should never have "http", it should always be "https".

The code could be improved to guarantee these things?